src/algebra/big_operators.lean

```diff
@@ -495,8 +495,27 @@ begin
   refl
 end
 
+@[to_additive]
+lemma prod_update_of_not_mem [decidable_eq α] {s : finset α} {i : α} (h : i ∉ s) (f : α → β) (b : β) :
+  s.prod (function.update f i b) = s.prod f :=
+begin
+  apply prod_congr rfl (λj hj, _),
+  have : j ≠ i, by { assume eq, rw eq at hj, exact h hj },
+  simp [this]
+end
+
+lemma prod_update_of_mem [decidable_eq α] {s : finset α} {i : α} (h : i ∈ s) (f : α → β) (b : β) :
+  s.prod (function.update f i b) = b * (s \ (singleton i)).prod f :=
+by { rw [update_eq_piecewise, prod_piecewise], simp [h] }
+
 end comm_monoid
 
+lemma sum_update_of_mem [add_comm_monoid β] [decidable_eq α] {s : finset α} {i : α}
+  (h : i ∈ s) (f : α → β) (b : β) :
+  s.sum (function.update f i b) = b + (s \ (singleton i)).sum f :=
+by { rw [update_eq_piecewise, sum_piecewise], simp [h] }
+attribute [to_additive] prod_update_of_mem
```

Comment:
Does `@[to_additive]` on `prod_update_of_mem` fail to produce a proof?

Reply:
Yes, with the error message `unknown declaration 'add_monoid.has_pow'`.

Reply:
Hmm, and that can't easily be fixed I guess. Because `has_pow` corresponds to some `has_smul`, but the order of arguments is swapped.